### PR TITLE
ActiveRecord typo fixes.

### PR DESCRIPTION
**Pull request by alkesh26**

### CODE REVIEW

File: activerecord/lib/active_record/associations/association.rb [hunk 1]
```diff
@@ -138,7 +138,7 @@ def extensions
       # +load_target+ unconditionally to get the \target.
       #
       # ActiveRecord::RecordNotFound is rescued within the method, and it is
-      # not reraised. The proxy is \reset and +nil+ is the return value.
+      # not re raised. The proxy is \reset and +nil+ is the return value.
```
Comment (vipulnsward) [inline activerecord/lib/active_record/associations/association.rb]: This is fine: https://en.wiktionary.org/wiki/reraise

File: activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ module ActiveRecord
   class ConnectionTimeoutError < ConnectionNotEstablished
   end
 
-  # Raised when a pool was unable to get ahold of all its connections
+  # Raised when a pool was unable to get hold of all its connections
```
Comment (vipulnsward) [inline activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb]: https://www.merriam-webster.com/dictionary/ahold

File: activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb [hunk 3]
```diff
@@ -31,7 +31,7 @@ def table_alias_for(table_name)
         table_name[0...table_alias_length].tr(".", "_")
       end
 
-      # Returns the relation names useable to back Active Record models.
+      # Returns the relation names usable to back Active Record models.
```
Comment (vipulnsward) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb]: https://www.dictionary.com/browse/useable

File: activerecord/lib/active_record/middleware/database_selector.rb [hunk 4]
```diff
@@ -43,7 +43,7 @@ def initialize(app, resolver_klass = Resolver, operations_klass = Resolver::Sess
 
       attr_reader :resolver_klass, :operations_klass
 
-      # Middleware that determines which database connection to use in a mutliple
+      # Middleware that determines which database connection to use in a multiple
```
Comment (vipulnsward) [inline activerecord/lib/active_record/middleware/database_selector.rb]: Can you keep just this and revert all other changes.

Comment (alkesh26) [reply activerecord/lib/active_record/middleware/database_selector.rb]: sure

**Pull request merged**